Reduce unsafe lambdas in WebKit/NetworkProcess & UIProcess

annevk opened the pull request:
#### cb9241b82712d2a3737b48e6fa6330a995a6612d
<pre>
Reduce unsafe lambdas in WebKit/NetworkProcess &amp; UIProcess
<a href="https://bugs.webkit.org/show_bug.cgi?id=291951">https://bugs.webkit.org/show_bug.cgi?id=291951</a>

Reviewed by Alex Christensen.

As per <a href="https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines">https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines</a>

Canonical link: <a href="https://commits.webkit.org/294431@main">https://commits.webkit.org/294431@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5a58b2890d2824aba2f2eb9816a3ea3623431f91

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/101792 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21460 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/11776 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/106950 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/52426 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/21768 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/29965 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77504 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34527 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/104799 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/16817 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/91920 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/57842 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/16643 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/51777 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/86498 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10015 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/109307 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/28925 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21301 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/86484 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-view-transitions/active-view-transition-on-non-root.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/29286 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88121 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86055 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/30809 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8531 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/23095 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16557 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/28853 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/34143 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/28664 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/31987 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/30223 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeEventDispatcher.mm:
```diff
@@ -204,8 +204,8 @@ void displayLinkFired(PlatformDisplayID displayID, DisplayUpdate, bool wantsFull
-    ScrollingThread::dispatch([dispatcher = Ref { *this }, wheelEvent, rubberBandableEdges] {
-        dispatcher->scrollingThreadHandleWheelEvent(wheelEvent, rubberBandableEdges);
+    ScrollingThread::dispatch([dispatcher = Ref { *this }, wheelEvent = WebWheelEvent { wheelEvent }, rubberBandableEdges] {
+        dispatcher->scrollingThreadHandleWheelEvent(CheckedRef { wheelEvent }.get(), rubberBandableEdges);
```
Why do we need to do this change?
AIUI, wheelEvent is captured by value so is safe for the lifetime of the lambda.

annevk replied:
https://build.webkit.org/results/Apple-Sequoia-Safer-CPP-Checks/294329%40main%20%283257%29/scan-build-output/StaticAnalyzer/WebKit/StaticAnalyzerReports/report-RemoteLayerTreeEventDispatcher.mm-unknown-0-ddd2d4.html#EndPath

cdumez commented on Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeEventDispatcher.mm:
```diff
@@ -204,8 +204,8 @@ void displayLinkFired(PlatformDisplayID displayID, DisplayUpdate, bool wantsFull
 
-    ScrollingThread::dispatch([dispatcher = Ref { *this }, wheelEvent, rubberBandableEdges] {
-        dispatcher->scrollingThreadHandleWheelEvent(wheelEvent, rubberBandableEdges);
+    ScrollingThread::dispatch([dispatcher = Ref { *this }, wheelEvent = WebWheelEvent { wheelEvent }, rubberBandableEdges] {
```
Do we really need the ` = WebWheelEvent { wheelEvent }` here? It should be identical to the previous code.

annevk committed:
Reduce unsafe lambdas in WebKit/NetworkProcess & UIProcess
https://bugs.webkit.org/show_bug.cgi?id=291951

Reviewed by Alex Christensen.

As per https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines

Canonical link: https://commits.webkit.org/294431@main
